evm-contracts/src/v0.6/dev/Flags.sol

```diff
@@ -0,0 +1,122 @@
+pragma solidity 0.6.6;
+
+
+import "../Owned.sol";
+import "../dev/SimpleAccessControl.sol";
+
+
+/**
+ * @title The Flags contract
+ * @notice Allows flags to signal to any reader on the access control list.
+ * The owner can set flags, or designate other addresses to set flags. The
+ * owner must turn the flags off, other setters cannot.
+ */
+contract Flags is Owned, SimpleAccessControl {
+
+  mapping(address => bool) private flags;
+  mapping(address => bool) private setters;
+
+  event FlagOn(
+    address indexed subject
+  );
+  event FlagOff(
+    address indexed subject
+  );
+  event SetterEnabled(
+    address indexed setter
+  );
+  event SetterDisabled(
+    address indexed setter
+  );
+
+  /**
+   * @notice read the warning flag status of a contract address.
+   * @param subject The contract address being checked for a warning.
+   */
+  function getFlag(address subject)
+    public
+    view
+    checkAccess()
+    returns (bool)
+  {
+    return flags[subject];
+  }
+
+  /**
+   * @notice allows owner to enable the warning flags for mulitple addresses.
+   * @param subjects List of the contract addresses whose flag is being raised
+   */
+  function setFlagsOn(address[] calldata subjects)
+    external
+    returns (bool)
+  {
+    require(msg.sender == owner || setters[msg.sender], "Only callable by enabled setters");
```

Comment:
It would be great if we extract this to a function modifier `_onlyWithWriteAccess`.

Reply:
I think the trend lately has been to avoid modifiers in favor of either 1) simply writing out the conditions, or 2) creating small helper functions containing the conditions (if they're used in more than one place).  What're your thoughts, @se3000?

Reply:
We used to only use modifiers, and there was a common complaint that it actually made the code harder to read. Right now the general rule I try to follow is that if we reuse a requirement then we pull it out into a modifier, but if it is only in one place it is more readable to leave it with the rest of the method. How does that sound to you @krebernisak?

Reply:
I am not going to lie, I ❤️ modifiers :)

But, I agree with you. I would not use them exclusively in every situation. It needs to be more readable, and/or needs to make sense for reuse.

There are even examples where a modifier is available like `onlyOwner`, but if I have a bunch of simple `requires` I would put the owner check in a `require(msg.sender == owner)` to improve readability, instead of having some checks as modifiers and some in the function body.

Here I suggested using a modifier because of the or condition. 

`_onlyWithWriteAccess` reads a lot better to me than:

```
require(msg.sender == owner || setters[msg.sender], "Only callable by enabled setters");
```

And it's even more complicated in the new version:

```
require(msg.sender == owner ||
  flaggingAccessController.hasAccess(msg.sender, msg.data),
  "No access");
```

If I need to know the details, I could inspect the `_onlyWithWriteAccess` modifier, but here I am bombarded with the details every time I read the `setFlagsOn`. I would argue that those details are not so important when reading the function, and it's enough for me to just know there is some write access modifier to check if interested.